torch/_inductor/fx_passes/micro_pipeline_tp.py

```diff
@@ -523,19 +639,19 @@ def fuse_all_gather_matmul(all_gather: _AllGatherMatch) -> None:
         or not torch.distributed.is_nccl_available()
     ):
         return
-
     c10d = torch.ops._c10d_functional
     from torch.distributed._symmetric_memory import (
         is_symm_mem_enabled_for_group,
         restride_A_shard_for_fused_all_gather_matmul,
     )
 
-    shard_node, ag_node, ag_res_node, gather_dim, group_name = (
+    shard_node, ag_node, ag_res_node, gather_dim, group_name, with_effects = (
```

Comment:
In this function, we don't care about the nodes between the `ag_node` (node whose target is all-gather) and the `ag_res_node`. Let's encapsulate the `with_effects` handling in `_find_consumer_matmuls()` and `_AllGatherMatch.replace_with()`..